neqo-transport/src/send_stream.rs

```diff
@@ -1063,61 +1115,290 @@ impl ::std::fmt::Display for SendStream {
 }
 
 #[derive(Debug, Default)]
-pub(crate) struct SendStreams(IndexMap<StreamId, SendStream>);
+pub struct OrderGroup {
+    // This vector is sorted by StreamId
+    vec: Vec<StreamId>,
+
+    // Since we need to remember where we were, we'll store the iterator next
+    // position in the object.  This means there can only be a single iterator active
+    // at a time!
+    next: usize,
+    // This is used when an iterator is created to set the start/stop point for the
+    // iteration.  The iterator must iterate from this entry to the end, and then
+    // wrap and iterate from 0 until before the initial value of next.
+    // This value may need to be updated after insertion and removal; in theory we should
+    // track the target entry across modifications, but in practice it should be good
+    // enough to simply leave it alone unless it points past the end of the
+    // Vec, and re-initialize to 0 in that case.
+}
+
+pub struct OrderGroupIter<'a> {
+    group: &'a mut OrderGroup,
+    // We store the next position in the OrderGroup.
+    // Otherwise we'd need an explicit "done iterating" call to be made, or implement Drop to
+    // copy the value back.
+    // This is where next was when we iterated for the first time; when we get back to that we stop.
+    started_at: Option<usize>,
+}
+
+impl<'a> OrderGroup {
+    pub fn iter_mut(&mut self) -> OrderGroupIter {
+        // Ids may have been deleted since we last iterated
+        if self.next >= self.vec.len() {
+            self.next = 0;
+        }
+        OrderGroupIter {
+            started_at: None,
+            group: self,
+        }
+    }
+
+    pub fn stream_ids(&mut self) -> &mut Vec<StreamId> {
+        &mut self.vec
+    }
```

Comment:
Does this really need to hand out a mutable reference to its underlying store?  What isn't possible if you do this?

Reply:
We need it for 
                    self.regular.stream_ids().push(stream_id);
and
        self.regular.stream_ids().clear();

Reply:
I could probably implement push() and clear() for OrderGroup().   that works.

Reply:
That's what I was working towards, yes.